### DISCUSSION TITLE
How can I position equation numbers to the left side?

### DISCUSSION THREAD

**Discussion by Complex2-Liu** [typst/typst#3106] [Q&A] [answered]
See [tex stackexchange 35497][post].

I try to use the [place] function, but the result is strange. When `alignment` is `left + bottom`, it seems like the number `(hello)` is located on the `right + top`. Why does the direction seem to be reversed? What is the parent container of an euqation number?

![haha](https://github.com/typst/typst/assets/40851978/8b771457-ceaf-4ff6-872e-a5604839007a)

The function `fixed-content-numbering` and `left-side-numbering` is defined as follows:

```typst
#let fixed-content-numbering(..nums) = {
  return "(hello)"
}

#let left-side-numbering(align: left, ..nums) = {
  place(align: align, "(hello)")
}
```

For now, my workaround is to use [move] in combination with [layout].

[post]: https://tex.stackexchange.com/questions/35497
[place]: https://typst.app/docs/reference/layout/place/
[move]: https://typst.app/docs/reference/layout/move/
[layout]: https://typst.app/docs/reference/layout/layout/

**Comment by Julian-Wassmann** [comment]
Hi,

Your approach with setting the `numbering` parameter of the equation doesn't work, because the numbering seems to be expected to be always on the right side (at least for text direction left-to-right).

Instead, the best option is to recreate the equation in a show rule. We can turn off the default numbering and provide our custom output, like this:
```typst
#show math.equation: eq => {
  // apply custom style only to block equations with numbering enabled
  if eq.block and eq.numbering != none {
    // default numbering of the equation
    let eqCounter = counter(math.equation).at(eq.location())
    let eqNumbering = numbering(eq.numbering, ..eqCounter)
  
    grid(
      // change "0pt" to "auto" to give the numbering its own space on the line
      columns: (0pt, 1fr),
      align(left + horizon)[#eqNumbering],
      // note that "numbering: none" avoids infinite recursion
      math.equation(eq.body, block: true, numbering: none),
    )
  } else {
    // default output for inline equations or without numbering
    eq
  }
}
```
Interestingly, setting the text direction to right-to-left also seems to do the job, but this is a dangerous approach, because it could result in unexpected behaviour.
```typst
#show math.equation: eq => {
  set text(dir: rtl)
  eq
}
```
Reproduction URL: https://typst.app/project/r9ely_g1bPta8rV9ohR0r0

**Reply by Complex2-Liu** [reply]
Thanks. Your solution is better than mine. It is better to use a grid rather than `move` and `layout`.

**Reply by Complex2-Liu** [reply, marked as answer]
As of typst 0.11.0, PR #3446, which provides `math.equation` with an argument `number-align`, has been merged.
